src/main/java/com/eerussianguy/blazemap/api/maps/IMapPart.java

```diff
@@ -0,0 +1,21 @@
+package com.eerussianguy.blazemap.api.maps;
+
+import com.eerussianguy.blazemap.api.BlazeRegistry;
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.Level;
+
+/**
+ * To unify the rendering behaviour of widgets that can take either a MapType or a Layer
+ */
+public interface IMapPart {
```

Comment:
Let's rename it `IClientComponent` maybe?
Best name I could come up with, but I'll accept a better one if you have it.

Reply:
Better name than what I came up with!

- [ ] Update interface name